pkg/client/repo/oci/oci.go

```diff
@@ -268,14 +268,35 @@ func (r *Repo) Fetch(name string, version string) (string, error) {
 
 // Has checks if a repo has a specific chart
 func (r *Repo) Has(name string, version string) (bool, error) {
-	versions, err := r.ListChartVersions(name)
+	ctx, cancel := context.WithTimeout(context.Background(), getTimeout)
+	defer cancel()
+
+	u := *r.url
+	// Form API endpoint URL from repo url
```

Comment:
can you please add a reference to the specification? Specifically to the [_Checking if content exists in the registry_](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#checking-if-content-exists-in-the-registry) section

Reply:
curiously, we added the initial implementation at the same time that the docs around this were merged (2y ago).

I remember we actually discussed the sub-optimal performance of this method due to not having a way of just checking for specific artifacts. It is nice that you took the time to look into it and fix it! 😄

Reply:
Should be done!

And you're welcome! We are using this application to sync our Helm charts, and it works great. Only this one method needed some work :)